qt/rest/qflightxmlclient/qflightxmlclient.h

```diff
@@ -0,0 +1,32 @@
+#ifndef QFLIGHTXMLCLIENT_H
+#define QFLIGHTXMLCLIENT_H
+
+#include <QObject>
+#include <QNetworkAccessManager>
+
+class QNetworkReply;
+class QAuthenticator;
+
+class QFlightXMLClient : public QObject
+{
+    Q_OBJECT
+public:
+    explicit QFlightXMLClient(QObject *parent = nullptr);
+    void airlineInfoRequest(const QString& airlineCode);
+
+private slots:
+    void serviceRequestFinished(QNetworkReply*);
+    void provideAuthenication(QNetworkReply *reply, QAuthenticator *ator);
+
+private:
+    QString fxmlUrl = "http://flightxml.flightaware.com/json/FlightXML3";
+    const QString Username = "slydog1";
```

Comment:
Please remove personal credentials.
This API key has now been revoked. Please return to the [FlightXML Key Generation](https://flightaware.com/commercial/flightxml/key) to get a new one.